config/application.rb

```diff
@@ -128,5 +129,15 @@ class Application < Rails::Application
 
     # Do not swallow errors in after_commit/after_rollback callbacks.
     # config.active_record.raise_in_transactional_callbacks = true
+
+    # Protect from "invalid byte sequence in UTF-8" errors,
+    # when a query or a cookie is a string with incorrect UTF-8 encoding.
+    config.middleware.insert_before(
+      Rack::Head,
+      Rack::UTF8Sanitizer,
+      sanitizable_content_types: [],
```

Comment:
Why are you doing `sanitizable_content_types: []`? 

Are you trying to clear the [defaults](https://github.com/whitequark/rack-utf8_sanitizer/blob/f2bcf865432a1470c30de3519df780a5b39cf9a2/lib/rack/utf8_sanitizer.rb#L69-L74)? 

If so why? Also I'm not sure setting it to [] will do anything will it??

Reply:
Yes, it clears the defaults.

I'm preventing [this function](https://github.com/whitequark/rack-utf8_sanitizer/blob/f2bcf865432a1470c30de3519df780a5b39cf9a2/lib/rack/utf8_sanitizer.rb#L117) from running, which would sanitize POST form data if not disabled.

Yes, setting it to [] works. [1](https://github.com/whitequark/rack-utf8_sanitizer/blob/f2bcf865432a1470c30de3519df780a5b39cf9a2/lib/rack/utf8_sanitizer.rb#L19) [2](https://github.com/whitequark/rack-utf8_sanitizer/blob/f2bcf865432a1470c30de3519df780a5b39cf9a2/lib/rack/utf8_sanitizer.rb#L120).

Reply:
Thanks for linking to the code. Does government-frontend POST any data?

Reply:
For government-frontend we only have one POST route: POST /*path(.:format)                 content_items#service_sign_in_options, but there are more in e.g. frontend and I wanted to use the same setup for all appls. I also wanted to apply the minimal change that resolves the issue, and sanitizing POST data adds unrelated additional stuff.

Reply:
It seems a bit hacky to pass in `[]` as a work around to prevent sanitize_rack_input from running, don't you think? It could easily stop working, and I'm not sure we'd know. If we really need to skip it, then at the very least I think we need an inline comment to explain this workaround. 

But in https://github.com/alphagov/transition/pull/1618 they've just gone with the default, and that app has lots of POST routes, so is this a valid concern?

Reply:
It's more of a question about what you want. Form data sanitisation is simply unrelated to the orginal problem, but if you think it's a nice to have feature, I think we can enable it. It's just another change, so maybe I can create a separate PR for removing that line, so the change can be easily reverted just in case anything goes wrong?

Reply:
That's a good idea 👍